hibernate-reactive-core/src/main/java/org/hibernate/reactive/pool/impl/ExternalSqlClientPool.java

```diff
@@ -82,4 +93,41 @@ public SqlExceptionHelper getSqlExceptionHelper() {
 	public CompletionStage<Void> getCloseFuture() {
 		return CompletionStages.voidFuture();
 	}
+
+
+	@Override
+	public CompletionStage<ResultSet> selectJdbcOutsideTransaction(String sql, Object[] paramValues) {
+		return preparedQueryOutsideTransaction( sql, Tuple.wrap( paramValues ) )
+				.thenApply( ResultSetAdaptor::new );
+	}
+
+	public CompletionStage<RowSet<Row>> preparedQueryOutsideTransaction(String sql, Tuple parameters) {
+		feedback( sql );
+		return getPool().preparedQuery( sql ).execute( parameters ).toCompletionStage()
+				.handle( (rows, throwable) -> convertException( rows, sql, throwable ) );
+	}
+
```

Comment:
Is there any reason for callers to call this instead of just... getting a connection from the pool and using it?

Reply:
I think that would work.

When I wrote the patch I felt it had the following benefit:
* I don't have to handle an open connection and make sure that it gets closed correctly in `ReactiveImprovedExtractionContextImpl`
* It's consistent with what we had before
* It's consistent with the Vert.x API, without exposing it. Not very important, but it helps us to use that existing method in Vert.x
* It's sending a stronger message about running in a different transaction (after sleeping on this, I don't know if I believe this any more. Still mentioning it for reference)

The only drawback I can see right now is a small amount of simple duplicated code and maybe a single method in the pool API that could be removed. If you feel strong about this or you have some concerned I haven't noticed, I can change the code.